cuda_core/cuda/core/experimental/_program.py

```diff
@@ -3,9 +3,361 @@
+    ptxas_options : str, optional
+        Specify options directly to ptxas, the PTX optimizing assembler.
+        Default: None
+        Maps to: --ptxas-options <options> (-Xptxas)
+    maxrregcount : int, optional
+        Specify the maximum amount of registers that GPU functions can use.
+        Default: None
+        Maps to: --maxrregcount=<N> (-maxrregcount)
+    ftz : bool, optional
+        When performing single-precision floating-point operations, flush denormal values to zero or preserve denormal values.
+        Default: False
+        Maps to: --ftz={true|false} (-ftz)
+    prec_sqrt : bool, optional
+        For single-precision floating-point square root, use IEEE round-to-nearest mode or use a faster approximation.
+        Default: True
+        Maps to: --prec-sqrt={true|false} (-prec-sqrt)
+    prec_div : bool, optional
+        For single-precision floating-point division and reciprocals, use IEEE round-to-nearest mode or use a faster approximation.
+        Default: True
+        Maps to: --prec-div={true|false} (-prec-div)
+    fmad : bool, optional
+        Enables (disables) the contraction of floating-point multiplies and adds/subtracts into floating-point multiply-add operations.
+        Default: True
+        Maps to: --fmad={true|false} (-fmad)
+    use_fast_math : bool, optional
+        Make use of fast math operations.
+        Default: False
+        Maps to: --use_fast_math (-use_fast_math)
+    extra_device_vectorization : bool, optional
+        Enables more aggressive device code vectorization in the NVVM optimizer.
+        Default: False
+        Maps to: --extra-device-vectorization (-extra-device-vectorization)
+    modify_stack_limit : bool, optional
+        On Linux, during compilation, use setrlimit() to increase stack size to maximum allowed.
+        Default: True
+        Maps to: --modify-stack-limit={true|false} (-modify-stack-limit)
+    dlink_time_opt : bool, optional
+        Generate intermediate code for later link-time optimization.
+        Default: False
+        Maps to: --dlink-time-opt (-dlto)
+    gen_opt_lto : bool, optional
+        Run the optimizer passes before generating the LTO IR.
+        Default: False
+        Maps to: --gen-opt-lto (-gen-opt-lto)
+    optix_ir : bool, optional
+        Generate OptiX IR. Only intended for consumption by OptiX through appropriate APIs.
+        Default: False
+        Maps to: --optix-ir (-optix-ir)
+    jump_table_density : int, optional
+        Specify the case density percentage in switch statements, and use it as a minimal threshold to determine whether jump table (brx.idx instruction) will be used to implement a switch statement.
+        Default: 101
+        Maps to: --jump-table-density=[0-101] (-jtd)
+    device_stack_protector : bool, optional
+        Enable (disable) the generation of stack canaries in device code.
+        Default: False
+        Maps to: --device-stack-protector={true|false} (-device-stack-protector)
```

Comment:
I think we should remove this for the time being. It's too advanced (and yet niche).